src/olympia/discovery/utils.py

```diff
@@ -16,10 +16,15 @@
 log = olympia.core.logger.getLogger('z.amo')
 
 
-def call_recommendation_server(telemetry_id):
+def call_recommendation_server(telemetry_id, locale, platform):
+    params = []
+    if locale:
+        params.append('locale=%s' % locale)
+    if platform:
+        params.append('platform=%s' % platform)
     endpoint = urlparse.urljoin(
         settings.RECOMMENDATION_ENGINE_URL,
-        '%s/' % telemetry_id)
+        '%s/%s%s' % (telemetry_id, '?' if params else '', '&'.join(params)))

```

Comment:
Instead of `'&'.join(params)` and the whole `params` list, I'd rather see a dict + `django.utils.http.urlencode`.

Reply:
https://github.com/mozilla/addons-server/pull/6916/commits/6df9381c64d2561c18e4e1ef5c4127282d5701f3